### PR TITLE
update spring-boot-dependencies bom to v3.3.0-RC1

### PR DESCRIPTION
**Pull request by mshima**
Follow https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.3-Release-Notes.

Fixes https://github.com/jhipster/generator-jhipster/issues/26205.

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/spring-boot/resources/spring-boot-dependencies.pom [hunk 1]
```diff
@@ -702,26 +635,96 @@
...
+      <dependency>
+        <groupId>org.flywaydb</groupId>
+        <artifactId>flyway-database-informix</artifactId>
+        <version>${flyway.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.flywaydb</groupId>
+        <artifactId>flyway-database-mongodb</artifactId>
+        <version>${flyway.version}</version>
+      </dependency>
```
Comment (mraible) [inline generators/spring-boot/resources/spring-boot-dependencies.pom:672]: Do we need these Flyway dependencies? We use Liquibase so they seem unnecessary.

Comment (mraible) [reply generators/spring-boot/resources/spring-boot-dependencies.pom:672]: Nevermind, I think this is just the Spring Boot dependencies POM that's copied into the project.

Comment (mshima) [reply generators/spring-boot/resources/spring-boot-dependencies.pom:672]: Yes, we have the spring-dependencies bom embedded so we can get information from it.
We use its properties.

**Comment by mshima** [comment]
Closing in favor of https://github.com/jhipster/generator-jhipster/pull/26242

**Pull request closed without merging**